packages/editor/src/client/panel/selectCommentsPanel.tsx

```diff
@@ -0,0 +1,240 @@
+import {ApolloError} from '@apollo/client'
+import EditIcon from '@rsuite/icons/legacy/Edit'
+import React, {useEffect, useMemo, useReducer, useState} from 'react'
+import {TFunction, useTranslation} from 'react-i18next'
+import {Link} from 'react-router-dom'
+import {
+  Button,
+  Checkbox,
+  Drawer,
+  Form,
+  IconButton,
+  Message,
+  Pagination,
+  Table,
+  toaster,
+  Toggle
+} from 'rsuite'
+
+import {FullCommentFragment, TagType} from '../api'
+import {useCommentListLazyQuery} from '../api/index'
+import {IconButtonTooltip} from '../atoms/iconButtonTooltip'
+import {PermissionControl} from '../atoms/permissionControl'
+import {SelectTags} from '../atoms/tag/selectTags'
+import {RichTextBlock} from '../blocks/richTextBlock/richTextBlock'
+import {CommentBlockValue} from '../blocks/types'
+import {DEFAULT_MAX_TABLE_PAGES, DEFAULT_TABLE_PAGE_SIZES} from '../utility'
+
+const onErrorToast = (error: ApolloError) => {
+  if (error?.message) {
+    toaster.push(
+      <Message type="error" showIcon closable duration={3000}>
+        {error?.message}
+      </Message>
+    )
+  }
+}
+
+const commentUsernameGenerator = (t: TFunction<'translation'>) => (comment: FullCommentFragment) =>
+  comment?.user
+    ? `${comment?.user.name}`
+    : ` ${comment?.guestUsername} ${t('comments.panels.unregisteredUser')}`
+
+export type SelectCommentPanelProps = {
+  itemId: string | null | undefined
+  selectedFilter: CommentBlockValue['filter']
+  onClose(): void
+  onSelect(filter: CommentBlockValue['filter'], comments: FullCommentFragment[]): void
+}
+
+export function SelectCommentPanel({
+  itemId,
+  selectedFilter,
+  onClose,
+  onSelect
+}: SelectCommentPanelProps) {
+  const [tagFilter, setTagFilter] = useState(selectedFilter.tags)
+  const [commentFilter, setCommentFilter] = useState(selectedFilter.comments)
+  const [allowCherryPicking, toggleCherryPicking] = useReducer(
+    cherryPicking => !cherryPicking,
+    !!commentFilter?.length
+  )
+  const [page, setPage] = useState<number>(1)
+  const [limit, setLimit] = useState<number>(10)
+  const {t} = useTranslation()
+  const [fetchComments, {data, loading}] = useCommentListLazyQuery({
+    onError: onErrorToast,
+    variables: {
+      filter: {
+        item: itemId
```

Comment:
Currently we only allow to get comments from the item itself. But this is just because the UI hasn't implemented this. The backend already allows to use any comment from any item.

I created a new issue to implement this: https://wepublish.atlassian.net/browse/WPC-888